src/euphorie/client/browser/login.py

```diff
@@ -89,9 +92,23 @@ def login(self, account, remember):
             )  # noqa: E501
 
     def transferGuestSession(self, account_id):
-        """Transfer session(s) from guest account to an existing user account"""
+        """Transfer session(s) from guest account to an existing user account.
+        Check that the account from which we want to transfer the session(s)
+        corresponds to the currently logged-in one (= guest session) as given via
+        the __ac cookie.
+        """
         if not account_id:
             return
+        cookie = self.request.cookies.get("__ac")
+        if not cookie:
+            return
+        ticket = decodebytes(safe_bytes(cookie))
+        try:
+            cookie_values = tktauth.splitTicket(ticket)
+        except ValueError:
+            return
+        if len(cookie_values) < 2 or cookie_values[1] != account_id:
+            return
         account = get_current_account()
```

Comment:
I am missing some details here, but that does not seem to be a valid fix...
Isn't it fixed by removing all the cruft and doing:
```
model.SurveySession.account_id == account.id
```
I am probably missing quite a lot of information...

Reply:
Never mind I think I got it... but still I do not understand why we are not using `account.id`